### PR TITLE
[WIP] Add default_{,floating_}border

### PR DESCRIPTION
**Pull request by clawoflight**
Implementation of i3/i3#2702

- [x] copy/rename `new_{window,float}` and associated states in `parser-config/config.spec`
- [x] rename `cfg_new_{window,float}` in `config_directives.c` and update calls in tests
- [x] extend the test case in `201-config-parser.t`
- [x] update the user guide

### CODE REVIEW

**Comment by clawoflight** [comment]
While working on this, I found that there was no test coverage of the state `DEFAULT_BORDER_PIXELS_PX` (e.g. `new_window pixel 10(px)?` -- should I add that?

**Comment by clawoflight** [comment]
I would welcome help getting the tests to run - they wouldn't run locally either, but that was _before doing any change_. Are the tests broken in `next`?

**Comment by Airblader** [comment]
> Are the tests broken in next?

No, AFAICT they are fine. Also, the failing tests make sense to me. You need to add the new config directives in one place so they are part of the expected output (which prints all config directives as possible options).

Apart from that and the userguide, this LGTM

File: testcases/t/201-config-parser.t [hunk 1]
```diff
@@ -266,19 +266,33 @@ $config = <<'EOT';
...
-cfg_new_window(new_window, normal, 2)
-cfg_new_window(new_window, none, -1)
-cfg_new_window(new_float, 1pixel, -1)
-cfg_new_window(new_float, normal, 2)
-cfg_new_window(new_float, none, -1)
-EOT
+cfg_default_border(new_window, 1pixel, -1)
+cfg_default_border(new_window, normal, 2)
+cfg_default_border(new_window, none, -1)
+cfg_default_border(default_border, 1pixel, -1)
```
Comment (clawoflight) [inline testcases/t/201-config-parser.t]: I added the new directives here though... Or did you mean something else?

**Comment by clawoflight** [comment]
It may be that I did not fix all tests, but my problem is that I couldn't get them to run *before* starting to work on this...

**Comment by orestisf1993** [comment]
The testsuite is documented: https://i3wm.org/docs/testsuite.html. Where did you have problems while following the instructions?

**Comment by clawoflight** [comment]
I never managed to run them without 1 failure, even before changing anything.
Specifically, this is what happens (on clean `next`):

[test-suite.log](https://github.com/i3/i3/files/1717658/test-suite.log)

**Comment by clawoflight** [comment]
Right, so I got the tests to run in CI - but I can still not run them locally.
This is what I get:

```

#   Failed test 'Error. You have specified 'C' as an Inline programming language.
# 
# I currently only know about the following languages:
#     Foo, foo
# 
# If you have installed a support module for this language, try deleting the
# config-x86_64-linux-thread-multi-5.026001 file from the following Inline DIRECTORY, and run again:
# 
#     /home/bennett/Github/i3/build/_Inline
# 
# (And if that works, please file a bug report.)
# 
#  at /home/bennett/Github/i3/build/../../i3/testcases/lib/i3test/XTEST.pm line 45.
# BEGIN failed--compilation aborted at /home/bennett/Github/i3/build/../../i3/testcases/lib/i3test/XTEST.pm line 45.
# Compilation failed in require at /home/bennett/Github/i3/build/../../i3/testcases/t/257-keypress-group1-fallback.t line 30.
# BEGIN failed--compilation aborted at /home/bennett/Github/i3/build/../../i3/testcases/t/257-keypress-group1-fallback.t line 30.
# '
#   at /home/bennett/Github/i3/build/../../i3/testcases/lib/TestWorker.pm line 53.
```

Any ideas? I thought I needed to install `perl-inline-c` (I'm on Arch), but that hasn't solved it.

**Comment by clawoflight** [comment]
I finally got them to run by trying random things such as deleting ~/perl5, ~/.cpan, and rebuilding the perl code. In the end, it worked after running make check (which failed), then complete-run.pl, which succeeded for the first time.
Now, make check seems to work consistently...

Maybe I should edit `docs/testsuite` to mention the inline c requirement?

**Comment by Airblader** [comment]
> Maybe I should edit docs/testsuite to mention the inline c requirement?

Sure, but let's do it in a separate PR! :-)

The changes LGTM now. Could you just squash them? Then we can go ahead and merge.

**Comment by orestisf1993** [comment]
cpanm finds the inline c requirement on my machine:

```
==> Found dependencies: Inline, X11::XCB, ExtUtils::PkgConfig, Inline::C
```

**Comment by clawoflight** [comment]
Squashed and force-pushed :-)

@orestisf1993 interesting... I did use cpan, could that be the difference?

**Comment by orestisf1993** [comment]
Still finds it here:

```
----      Unsatisfied dependencies detected during      ----
    ExtUtils::PkgConfig [requires]
    Inline [requires]
    Inline::C [requires]
    X11::XCB [requires]
```

**Comment by clawoflight** [comment]
Interesting...

**Comment by clawoflight** [comment]
@Airblader This is done, not sure what to do about Inline::C though. I can run tests locally now, so maybe I'll just keep that in the back of my head until someone else has that problem.

**Comment by clawoflight** [comment]
@Airblader This is done, not sure what to do about Inline::C though. I can run tests locally now, so maybe I'll just keep that in the back of my head until someone else has that problem.

**Comment by Airblader** [comment]
Thanks!

**Pull request merged**